frontend/tests/e2e/pageObjects/studentAdmin/institute/authoritiesDetailsPage.js

```diff
@@ -4,12 +4,75 @@ const log = require('npmlog')
 class AuthoritiesDetailsPage {
   constructor() {
     this.authorityNameField = Selector('#authorityName');
+    this.editAuthorityButton = Selector('#editButton');
+    this.saveAuthorityEditButton = Selector('#saveButton');
+    this.authorityPhoneNumberInput = Selector('#phoneNumberField');
+    this.authroityPhoneNumberSpan = Selector('span');
+    this.authorityEmailInput = Selector('#emailField');
+    this.authorityMAddressLine1 = Selector('#mailAddressLine1');
+    this.authorityMAddressCity = Selector('#mailAddressCity');
+    this.authorityMAddressProv = Selector('#mailAddressProvince');
+    this.authorityMAddressCountry = Selector('#mailAddressCountry');
+    this.authorityMAddressPostal = Selector('#mailAddressPostal');
+    this.authorityEmailSpan = Selector('span');
   }
 
   async verifyAuthorityNumberAndName(name) {
     await t.expect(this.authorityNameField.withText(name).count).eql(1, {timeout: 3000});
     log.info('Authority detail name and number verified');
   }
+
+  async verifyAuthorityPhoneNumber(){
+    await t.expect(this.authroityPhoneNumberSpan.withText('101-101-1100').innerText).contains('101-101-1100');
+    log.info('Authority Phone Number Verified');
+  }
+
+  async verifyAuthorityEmailAddress(){
+    await t.expect(this.authorityEmailSpan.withText('test2@test.com').innerText).contains('test2@test.com');
+    log.info('Authority Email Verified');
+  }
+
+
+  async clickEditAuthorityButton(){
+    await t.click(this.editAuthorityButton);
+    log.info("Edit Authority Contact Button Clicked");
+  }
+
+  async clickSaveAuthorityEditButton(){
+    await t.click(this.saveAuthorityEditButton);
+    log.info("Save Authority Contact Button Clicked");
+  }
+
+  async editAuthorityDetails(){
+    await t.typeText(this.authorityPhoneNumberInput, '1011011100', {replace: true});
+    await t.typeText(this.authorityEmailInput, 'test2@test.com', {replace: true});
+    await t.typeText(this.authorityMAddressLine1, '123 Maple Dr', {replace: true})
+    await t.typeText(this.authorityMAddressCity, 'Victoria', {replace: true})
+    await this.selectMAddressProvince();
+    await this.selectMAddressCountry();
+    await t.typeText(this.authorityMAddressPostal, 'a1b2b3', {replace: true});
+
+
+    log.info("Authority Details Edit Complete");
+  }
+
```

Comment:
Could you verify that the selectors for the Mailing Address Province and Country are working correctly? I'm noticing it takes a long time to select.